src/duk_error_throw.c

```diff
@@ -107,10 +113,11 @@ DUK_INTERNAL void duk_err_create_and_throw(duk_hthread *thr, duk_errcode_t code)
 	 *  Finally, longjmp
 	 */
 
-	thr->heap->handling_error = 0;
-
 	duk_err_setup_heap_ljstate(thr, DUK_LJ_TYPE_THROW);
 
+	thr->callstack_max -= DUK_CALLSTACK_DEFAULT_MAX * 2;  /* reset reclimit */
+	thr->heap->handling_error = 0;
```

Comment:
Technically it might be enough just to set it back to DUK_CALLSTACK_DEFAULT_MAX directly and forgo the assert entirely (the entry assert would catch it on the second go-around if it were skipped).